java-checks/src/main/java/org/sonar/java/checks/helpers/SyntaxNodePredicates.java

```diff
@@ -38,7 +40,25 @@ public boolean apply(Tree syntaxNode) {
     };
   }
 
-  public static Predicate<String> isSubtypeOf(final Type type) {
+  public static Predicate<Tree> kinds(final Set<Tree.Kind> kinds) {
+    return new Predicate<Tree>() {
+      @Override
+      public boolean apply(Tree input) {
+        return kinds.contains(input.kind());
+      }
+    };
+  }
+
+  public static Predicate<String> typeIs(final Type type) {
+    return new Predicate<String>() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this a named "static" inner class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)